src/parpeoptimization/optimizationProblem.cpp

```diff
@@ -54,6 +54,87 @@ void *getLocalOptimumThreadWrapper(void *optimizationProblemVp) {
     return result;
 }
 
+void optimizationProblemGradientCheckMultiEps(OptimizationProblem *problem,
+                                      int numParameterIndicesToCheck
+                                      ) {
+    // set eps
+    std::vector<double> multi_eps {1e-1, 1e-3, 1e-5, 1e-7, 1e-9};
+
+    // setting the number of parameters to the minimum of
+    // numParamaterIndicesToCheck and dimension of the problem
+    int numParameters = problem->cost_fun_->numParameters();
+    numParameterIndicesToCheck =
+        std::min(numParameterIndicesToCheck, numParameters);
```

Comment:
Isn't it assigned here but at the same time also read here? Since it is assigned the minimum of what is given as input and the total number of parameters.

Reply:
It is, but what is it used for afterwards?